File: iOS/Objective-C/VioozerIntergrationDemoApp/VioozerIntergrationDemoApp/ViewController.m
//
//  ViewController.m
//  VioozerIntergrationDemoApp
//
//  Created by Yedidya Reiss on 17/01/2017.
//  Copyright © 2017 Yedidya Reiss. All rights reserved.
//

#import "ViewController.h"
#import <VioozerAPI.h>
@interface ViewController ()

@property (weak,nonatomic) IBOutlet UILabel* lblSDKState;
@property (weak,nonatomic) IBOutlet UIView* viewBaseLine;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    
    [[NSNotificationCenter defaultCenter] addObserver:self
                                             selector:@selector(vioozerReady)
                                                 name:VIOOZER_IS_READY_NOTIFICATION object:nil];
    
    [[NSNotificationCenter defaultCenter] addObserver:self
                                             selector:@selector(vioozerFailed)
                                                 name:VIOOZER_FAILED_NOTIFICATION object:nil];
    
}


-(void) vioozerReady {
    [[NSOperationQueue mainQueue] addOperationWithBlock:^{
        _lblSDKState.text = @"INITIALIZATION OK";
        _lblSDKState.textColor = [UIColor greenColor];
        _viewBaseLine.backgroundColor = [UIColor greenColor];
    }];
    
}

-(void) vioozerFailed {
    [[NSOperationQueue mainQueue] addOperationWithBlock:^{
        _lblSDKState.text = @"INITIALIZATION ERROR";
        _lblSDKState.textColor = [UIColor redColor];
        _viewBaseLine.backgroundColor = [UIColor redColor];
    }];
}


- (void)dealloc {
    [[NSNotificationCenter defaultCenter] removeObserver:self];
}

- (IBAction) presentQuestion {
    
    // name, type and address are free-text fields.
    VioozerLocation * l = [VioozerLocation VioozerLocationWithCoordinaes:CLLocationCoordinate2DMake (37.759734, -122.427063)
                                                                    name:@"Dolores Park"
                                                                    type:@"Park"
                                                              andAddress:@"19th & Dolores St, San Francisco, CA 94114, USA"];
    
    
    [VIOOZER_API showAskQuestionOnLocations:@[l] inCategories:@[]];
}


-(IBAction)presentVioozerMyAvtivity {
    [VIOOZER_API openMyActivity];
}

@end

File: iOS/Swift/VioozerSwiftDemoApp/VioozerSwiftDemoApp/ViewController.swift
//
//  ViewController.swift
//  VioozerSwiftDemoApp
//
//  Created by Yedidya Reiss on 17/01/2017.
//  Copyright © 2017 Yedidya Reiss. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    
    @IBOutlet weak var lblSDKState : UILabel!
    @IBOutlet weak var viewBaseLine : UIView!

    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        NotificationCenter.default.addObserver(self, selector: #selector(ViewController.vioozerIsReady), name: NSNotification.Name(rawValue: VIOOZER_IS_READY_NOTIFICATION), object: nil)
        
        NotificationCenter.default.addObserver(self, selector: #selector(ViewController.vioozerFailed), name: NSNotification.Name(rawValue: VIOOZER_FAILED_NOTIFICATION), object: nil)
        
        
    }
    
    
    deinit {
        NotificationCenter.default.removeObserver(self)
    }

    func vioozerIsReady() {
        OperationQueue.main.addOperation {
            self.lblSDKState.text = "INITIALIZATION OK"
            self.lblSDKState.textColor = UIColor.green
            self.viewBaseLine.backgroundColor = UIColor.green
        }
    }
    
    func vioozerFailed() {
        OperationQueue.main.addOperation {
            self.lblSDKState.text = "INITIALIZATION ERROR"
            self.lblSDKState.textColor = UIColor.red
            self.viewBaseLine.backgroundColor = UIColor.red
        }
    }

    @IBAction func presentQuestion() {
        let c = CLLocationCoordinate2D(latitude: 37.759734, longitude: -122.427063)
        
        if let l = VioozerLocation(coordinaes: c , name: "Dolores Park", type: "Park", andAddress: "19th & Dolores St, San Francisco, CA 94114, USA") {
            
            VioozerAPI.sharedInstance().showAskQuestion(onLocations: [l], inCategories: [])
        }
    }
    
    
    @IBAction func presentVioozerActivity() {
        VioozerAPI.sharedInstance().openMyActivity()
    }
    

}

